src/ZenithAngleCalc.jl

```diff
@@ -1,72 +1,68 @@
-    Δt = -FT(2)*e*sin(MA) + tan(γ/FT(2))^FT(2)*sin(FT(2)*(MA + ϖ)) 
-    Δt = mod(Δt+π, FT(2)*FT(π))-π
+    _Δt = -2*e*sin(MA) + tan(γ/2)^2*sin(2*(MA + ϖ));
+    return mod(_Δt+FT(π), FT(2π)) - FT(π)
 end
 
 # calculate the distance, declination, and hour angle (at lon=0)
-function distance_declination_hourangle(::Type{FT}, date::DateTime, param_set::APS, eot_correction::Bool) where {FT <: Real}
+function distance_declination_hourangle(::Type{FT},
+                                        date::DateTime,
+                                        param_set::APS,
+                                        eot_correction::Bool,
+                                        milankovitch::Bool) where {FT <: Real}
     Ya::FT = year_anom(param_set)
     day_length::FT = Planet.day(param_set)
     AU::FT = astro_unit()
-
     _epoch::FT = epoch(param_set)
     M0::FT = mean_anom_epoch(param_set)
-    ϖ0::FT = lon_perihelion_epoch(param_set)
 
-    γ::FT = obliq_epoch(param_set)
-    ϖ::FT = lon_perihelion(param_set)
-    e::FT = eccentricity_epoch(param_set)
+    t0_years = _epoch / Ya; # s --> years
+    t_years = FT(datetime2julian(date)) * (day_length / Ya); # DateTime --> years
     
-    # time of vernal equinox in the epoch (rearrangement of 3.6 and 3.10)
-    M_v0 = mean_anomaly_vernal_equinox(ϖ0, e)
-    time_v = Ya * (M_v0 - M0) / (FT(2)*FT(π)) + _epoch
-
-    # mean anomaly given mean anomaly at vernal equinox (3.10)
-    time = datetime2julian(date) * day_length # in seconds
-    M_v = mean_anomaly_vernal_equinox(ϖ, e)
-    MA = mod(FT(2)*FT(π) * FT(time - time_v) / Ya + M_v, FT(2)*FT(π))
+    # mean anomaly given mean anomaly at epoch (3.6)
+    MA = mod(FT(2π) * (t_years - t0_years) + M0, FT(2π))
+
+    # calculate orbital parameters or take values at J2000
+    if milankovitch
+        ϖ = FT(ϖ_spline(t_years - t0_years));
+        γ = FT(γ_spline(t_years - t0_years));
+        e = FT(e_spline(t_years - t0_years));
+    else
+        ϖ = FT(lon_perihelion_epoch(param_set));
+        γ = FT(obliq_epoch(param_set));
+        e = FT(eccentricity_epoch(param_set));
+    end
 
     # true anomaly, radians (3.8)
     TA = true_anomaly(MA, e)
 
     # true longitude, radians (3.9)
-    TL = mod(TA + ϖ, FT(2)*FT(π))
+    TL = mod(TA + ϖ, FT(2π))
 
     # declination, radians (3.16)
-    δ = mod(asin(sin(γ) * sin(TL)), FT(2)*FT(π))
+    δ = mod(asin(sin(γ) * sin(TL)), FT(2π))
 
     # earth-sun distance, (3.1)
-    d = AU * (FT(1) - e^FT(2)) / (FT(1) + e*cos(TA))
+    d = AU * (1 - e^2) / (1 + e*cos(TA))
 
     # hour angle, zero at local solar noon, radians (3.17)
     if eot_correction
-        Δt = equation_of_time(e, MA, γ, ϖ) / 2π * day_length # radians to seconds
+        Δt = equation_of_time(e, MA, γ, ϖ) / FT(2π) * day_length # radians to seconds
     else
         Δt = FT(0)
     end
-    η_UTC = mod(FT(2)*FT(π) * FT(time + Δt) / day_length, FT(2)*FT(π))
+    t0 = FT(mod(datetime2julian(date), 1.0)) * day_length; # s in this day
```

Comment:
Why t0 here? It seems confusing because t0 previously referred to time of epoch, but not here (I believe).